pygmt/base_plotting.py

```diff
@@ -141,6 +141,19 @@ def coast(self, **kwargs):
 
         """
         kwargs = self._preprocess(**kwargs)
+        if not (
+            "C" in kwargs
+            or "G" in kwargs
+            or "S" in kwargs
+            or "I" in kwargs
+            or "N" in kwargs
+            or "Q" in kwargs
+            or "W" in kwargs
+        ):
```

Comment:
I'm wondering if there is a better and more compact way for this. 

I wrote some functions (https://github.com/GenericMappingTools/pygmt/commit/313560e78cc05414f3e7dd43e2b1aca68dd44c37, unmerged yet) three years ago, but they're not satisfactory.

Reply:
Rather than putting all of the error handling in another function, maybe we can create an `args_in_kwargs` function that takes the list of required aliases and the kwargs list and uses a for loop to determine if any of the aliases are in the kwargs list. It would return a Boolean value, and then  `coast()` in `base_plotting.py` uses an if statement to determine if it should raise an error.

Reply:
> Rather than putting all of the error handling in another function, maybe we can create an `args_in_kwargs` function that takes the list of required aliases and the kwargs list

Sounds a good idea.

Reply:
@seisman Do you want me to make this `args_in_kwargs` function a separate pull request? I think that's the better move for now, as there are some `GMTInvalidInput Errors` already in `base_plotting.py`.

Reply:
> Do you want me to make this `args_in_kwargs` function a separate pull request?

Yes, please.

Reply:
Okay; should I close this PR then?

Reply:
I think you can keep it open, and get back to this PR after finishing the "args_in_Kwargs" function.

Reply:
> I'm wondering if there is a better and more compact way for this.

There is:

```suggestion
        if not any(arg in kwargs for arg in ["C", "G", "S", "I", "N", "Q", "W"]):
```

> I wrote some functions ([313560e](https://github.com/GenericMappingTools/pygmt/commit/313560e78cc05414f3e7dd43e2b1aca68dd44c37), unmerged yet) three years ago, but they're not satisfactory.

Do we want to raise the GMTInvalidInput error in `fig.coast`, or in the `arg_in_kwargs` function being done at #791? I feel like the error handling code at 313560e78cc05414f3e7dd43e2b1aca68dd44c37 is pretty good and could probably be modified and go in #791.

Reply:
> Do we want to raise the GMTInvalidInput error in `fig.coast`, or in the `arg_in_kwargs` function

We have to raise the error in `fig.coast`, as the `arg_in_kwargs` function doesn't know the long-form aliases.

> I feel like the error handling code at [313560e](https://github.com/GenericMappingTools/pygmt/commit/313560e78cc05414f3e7dd43e2b1aca68dd44c37) is pretty good and could probably be modified and go in #791.

Do you mean the `GMTOptionError` exception?

Reply:
I briefly looked at trying to have the function raise the error, but I couldn't figure out how to pass the aliases to the function to translate between the single character names and the Python arguments.

Reply:
> We have to raise the error in `fig.coast`, as the `arg_in_kwargs` function doesn't know the long-form aliases.

True. It will be nice to have a generalizable function but it's fine to hardcode the GMTInvalidInput error here in `fig.coast` for now. Would be good to open an issue to keep track of this.

> Do you mean the `GMTOptionError` exception?

Yes, but for a separate discussion/PR :smile: